src/backend/utils/adt/ruleutils.c

```diff
@@ -11897,7 +11897,7 @@ generate_relation_name(Oid relid, List *namespaces)
 	if (!need_qual)
 		need_qual = !RelationIsVisible(relid);
 
-	if (need_qual)
+	if (need_qual || sql_dialect == SQL_DIALECT_TSQL)
```

Comment:
Yanjie's reply:

This line is to fix https://jira.rds.a2z.com/browse/BABEL-5284:
During pg_get_querydef (pg deparse function), PG will call `generate_relation_name(Oid relid, List *namespaces)` to generate rel name. PG will skip table schema name when the calling env is using the same schema name. PG uses this function only during pg_dump, and pg_get_ruledef (when check view definition). So for pivot view, when the current schema name is equal to (search path includes current schema name) the relation's schema name in the view definition,  PG will skip the schema name in the view definition, and this will result table not found issue.
The fix is also straightforward, we can add a sql_dialect check, when it is tsql, we always produce fully qualified name. I tested in my local environment and it worked great.
```
	/* Otherwise, qualify the name if not visible in search path */
	if (!need_qual)
		need_qual = !RelationIsVisible(relid);
```
```
	if (need_qual || sql_dialect == SQL_DIALECT_TSQL)
		nspname = get_namespace_name_or_temp(reltup->relnamespace);
	else
		nspname = NULL;
```
fix is in src/backend/utils/adt/ruleutils.c:12110